modules/core/src/commonMain/kotlin/fr/acinq/lightning/payment/Bolt11Invoice.kt

```diff
@@ -11,16 +11,21 @@ import fr.acinq.lightning.*
 import fr.acinq.lightning.Lightning.randomBytes32
 import fr.acinq.lightning.utils.*
 import fr.acinq.lightning.wire.LightningCodecs
+import fr.acinq.secp256k1.Hex
 import kotlin.experimental.and
 
 data class Bolt11Invoice(
```

Comment:
A comment is absolutely required for such non-conventional implementation.

```suggestion
/**
 * @param encodedAmount We counterintuitively store the string-serialized amount instead of the numeric value. The goal is to preserve the
 *                      validity of the signature for invoices generated by 3rd party software when the amount is not canonically encoded.
 */
data class Bolt11Invoice(
```